src/main/java/com/thebois/models/beings/AbstractBeingGroup.java

```diff
@@ -2,13 +2,14 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * An abstract implementation of IBeingGroup.
  */
 public abstract class AbstractBeingGroup implements IBeingGroup {
 
-    private Collection<IBeing> beings = new ArrayList<>();
+    private final ConcurrentLinkedQueue<IBeing> beings = new ConcurrentLinkedQueue<>();
```

Comment:
Why concurrent? We're not accessing it concurrently, are we?

And why a queue? It's just the beings in a being group, they shouldn't be queued up, right?

A `Collection` is just fine.

Reply:
No libGDX uses multiple threads and breaks it, so it needs to be concurrent

Reply:
That's strange, we never encountered this before. From which thread is this being called?

Reply:
You should be able to see it when running the debugger, in one of the panes. 

Yeah, that's fair, if it doesn't work without this then I suppose it's necessary. What worries me is that this thread-issue suddenly arose without a clear origin. What exactly is running on a separate thread and calling into our model, should it even be calling the model if it is on a separate thread? 

The fact that the model is even being called from another thread could cause issues, because the entire model is written single-threadedly. Sure, in this case solving it was as simple as swapping the list-type, but more difficult-to-debug concurrency issues could arise.

Anyway, we don't really have the time to find this out now, so you're right let's just leave it as-is.